saw-core-what4/src/Verifier/SAW/Simulator/What4.hs

```diff
@@ -1128,6 +1128,39 @@ w4EvalBasic sym sc m addlPrims ref unintSet t =
      cfg <- Sim.evalGlobal' m (give sym constMap `Map.union` addlPrims) extcns uninterpreted
      Sim.evalSharedTerm cfg t
 
+
+-- | Evaluate a saw-core term to a What4 value for the purposes of
+--   using it as an input for symbolic simulation.  This will evaluate
+--   primitives, but will cancel evaluation and return the associated
+--   @NameInfo@ if it encounters a constant value with an @ExtCns@
```

Comment:
More haddock nitpicks: `@NameInfo@` and `@ExtCns@` would be better as `'NameInfo'` and `'ExtCns'` because they are declared identifiers